subsys/settings/src/settings_shell.c

```diff
@@ -151,11 +215,25 @@ static int cmd_delete(const struct shell *shell_ptr, size_t argc, char *argv[])
 }
 
 SHELL_STATIC_SUBCMD_SET_CREATE(settings_cmds,
-			       SHELL_CMD_ARG(list, NULL, "[<subtree>]", cmd_list, 1, 1),
-			       SHELL_CMD_ARG(read, NULL, "<name>", cmd_read, 2, 0),
-			       SHELL_CMD_ARG(write, NULL, "<name> <hex>", cmd_write, 3, 0),
-			       SHELL_CMD_ARG(delete, NULL, "<name>", cmd_delete, 2, 0),
-			       SHELL_SUBCMD_SET_END);
+	SHELL_CMD_ARG(list, NULL,
+		"List all settings in a subtree (omit to list all)\n"
+		"Usage: settings list [<subtree>]",
+		cmd_list, 1, 1),
+	SHELL_CMD_ARG(read, NULL,
+		"Read a specific setting\n"
+		"Usage: settings read [type] <name>\n"
```

Comment:
Following the existing format for `settings list`:
```suggestion
		"Usage: settings read [<type>] <name>\n"
```

Reply:
The settings list was actually wrong, it wasn't following the POSIX convention. I changed it to [subtree]
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html